components/Ripple/src/private/MDCRippleLayer.m

```diff
@@ -164,6 +164,8 @@ - (void)endRippleAnimated:(BOOL)animated completion:(MDCRippleCompletionBlock)co
   fadeOutAnim.beginTime = [self convertTime:_rippleTouchDownStartTime + delay fromLayer:nil];
   fadeOutAnim.timingFunction =
       [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionLinear];
+  fadeOutAnim.fillMode = kCAFillModeForwards;
+  fadeOutAnim.removedOnCompletion = NO;
```

Comment:
Just noticed this change as part of https://github.com/material-components/material-components-ios/pull/6837. While this change does work, the ideal Core Animation practice for getting animations to "stick" once completed is to update the model layer value of the affected layer.

So in this case:

```objc
[CATransaction begin];
[CATransaction setDisableActions:YES];
self.alpha = 0;
[CATransaction commit];
```

You can then add the animation as specified here without requiring a fill mode.

Reply:
@codeman7 @yarneo for visibility here.

Reply:
Thanks Jeff, I was aware of this best practice, if you'll see above: https://github.com/yarneo/material-components-ios/blob/798579947d3269e1d42eeac7b0e16a28bc8b8821/components/Ripple/src/private/MDCRippleLayer.m#L60-L61. However, I wasn't aware of setDisableActions so I had tried it without that and noticed the same blinking still occurring. I can try the proposed way as a future follow up.

Reply:
Ah yep; without `setDisableActions:YES` there's a decent chance your layer will add an implicit animation that can end up conflicting with an explicitly added animation.

The key to debugging animations in these cases is to dump the layer's animations at the end of the function, usually with a `po layer`. If you see any animations that you didn't add, there's a chance an implicit animation was accidentally created somewhere.

Reply:
From your code snippet, wouldn't you want to do:
```
[CATransaction setCompletionBlock:^{
    self.alpha = 0;
}];
```

Wouldn't you want to do the desired outcome in a completion block?